lib/OpenQA/Worker.pm

```diff
@@ -750,18 +751,22 @@ sub _handle_job_status_changed ($self, $job, $event_data) {
     }
 }
 
-sub _load_avg ($field = 2) {
-    my $value = eval { (split(' ', path($ENV{OPENQA_LOAD_AVG_FILE} // '/proc/loadavg')->slurp))[$field] };
+sub _load_avg () {
+    my @load = eval { split(' ', path($ENV{OPENQA_LOAD_AVG_FILE} // '/proc/loadavg')->slurp) };
+    splice @load, 3;    # remove non-load numbers
     log_warning "Unable to determine average load: $@" if $@;
-    return looks_like_number($value) ? $value : undef;
+    return undef unless all { *looks_like_number } @load;
+    return @load;
 }
 
 sub _check_system_utilization ($self) {
     my $settings = $self->settings->global_settings;
     return undef unless my $threshold = $settings->{CRITICAL_LOAD_AVG_THRESHOLD};
-    my $load_avg = _load_avg;
-    return "The average load $load_avg is exceeding the configured threshold of $threshold."
-      if defined $load_avg && $load_avg >= $threshold;
+    my @load = _load_avg;
+    return undef unless @load;
```

Comment:
if you `return undef` explicitly in `_load_avg`, `@load` will always contain at least 1 element here, so this will never return here.